rundown/rundown.py

```diff
@@ -0,0 +1,433 @@
+from typing import Union, List, Dict, Optional
+
+import requests
+from pydantic import parse_obj_as
+
+from rundown.utils import utc_shift
+from rundown.resources.sportsbook import Sportsbook
+from rundown.resources.sport import Sport
+from rundown.resources.date import Date, Epoch
+from rundown.resources.team import Team
+from rundown.resources.events import Events
+from rundown.resources.event import Event, EventLinePeriods
+from rundown.resources.line import Moneyline, Spread, Total
+from rundown.resources.lineperiods import LinePeriods
+from rundown.resources.schedule import Schedule
+from rundown.usercontext import user_context
+
+
+class _RapidAPIAuth:
+    """Authorization configuration required for making requests to RapidAPI."""
+
+    def __init__(self, api_key: str):
+        self.api_key = api_key
+        self.api_host = "therundown-therundown-v1.p.rapidapi.com"
```

Comment:
Changed to `_RundownBase` and `_RapidAPIBase`. I don't see anything about versions in the documentation, would this just be handling future version upgrades? I could add a parameter to `Rundown` which allows the user to specify which API version they are using, e.g. 'v2', which would get passed to the base classes and the configuration would be changed accordingly.

Reply:
>  I don't see anything about versions in the documentation, would this just be handling future version upgrades?

From what I can tell, it is actually the `x-rapidapi-host` header that may be used to specify the version given the current contains the word `v1`, although I could be wrong. Because of that, we probably don't need a version, or, we could try to map the version in the configuration to the `api_host` value. 

What do you think? I wish this were more clear on RapidAPI's part.